MoreCyclopsUpgrades/Monobehaviors/CyBioReactorMono.cs

```diff
@@ -0,0 +1,446 @@
+                    canAdd = true;
+                }
+            }
+
+            if (!canAdd && verbose)
+            {
+                ErrorMessage.AddMessage(Language.main.Get("BaseBioReactorCantAddItem"));
+            }
+
+            return canAdd;
+        }
+
+        private bool IsAllowedToRemove(Pickupable pickupable, bool verbose) => false;
+
+        private float ProducePower(float powerDrawnPerItem)
+        {
+            float powerProduced = 0f;
+
+            if (powerDrawnPerItem > 0f && // More than zero energy being produced per item per time delta
+                this.MaterialsProcessing.Count > 0) // There should be materials in the reactor to process
+            {
+                foreach (BioEnergy material in this.MaterialsProcessing)
+                {
+                    float availablePowerPerItem = Mathf.Min(material.RemainingEnergy, material.Size * powerDrawnPerItem);
+
+                    material.RemainingEnergy -= availablePowerPerItem;
+                    powerProduced += availablePowerPerItem;
+
+                    if (material.FullyConsumed)
+                        this.FullyConsumed.Add(material);
+                }
+            }
+
+            if (this.FullyConsumed.Count > 1)
+            {
+                foreach (BioEnergy material in this.FullyConsumed)
+                {
+                    this.MaterialsProcessing.Remove(material);
+                    this.Container.RemoveItem(material.Pickupable, true);
+                    Destroy(material.Pickupable.gameObject);
+                }
+
+                this.FullyConsumed.Clear();
+            }
+
+            return powerProduced;
+        }
+
+        private void UpdateDisplayText()
+        {
+            if (textDelay-- > 0)
+                return; // Slow down the text update
+
+            textDelay = TextDelayTicks;
+
+            foreach (BioEnergy material in this.MaterialsProcessing)
+                material.UpdateInventoryText();
+        }
+
+        public float Constructed
+        {
+            get => _constructed;
+            set
+            {
+                value = Mathf.Clamp01(value);
+                if (_constructed != value)
+                {
+                    _constructed = value;
+                    if (_constructed < 1f)
+                    {
+                        if (_constructed <= 0f)
+                        {
+                            Destroy(this.gameObject);
+                        }
+                    }
+                }
+            }
+        }
+
+        public void OnProtoSerialize(ProtobufSerializer serializer)
+        {
+            SaveData.ReactorBatterCharge = this.Battery.charge;
+            SaveData.SaveMaterialsProcessing(this.MaterialsProcessing);
+
+            SaveData.Save();
+        }
+
+        public void OnProtoDeserialize(ProtobufSerializer serializer)
+        {
+            isLoadingSaveData = true;
+
+            InitializeStorageRoot();
```

Comment:
Is there anything else that should be initialized in the `OnProtoDeserialize` event?